File: sdks/csharp/src/NovaBilling/Types/PaginatedPlanOverrideResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record PaginatedPlanOverrideResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("data")]
    public IEnumerable<PlanOverrideResponse> Data { get; set; } = new List<PlanOverrideResponse>();

    [JsonPropertyName("meta")]
    public Dictionary<string, object?> Meta { get; set; } = new Dictionary<string, object?>();

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/paginated_plan_override_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class PaginatedPlanOverrideResponse < Internal::Types::Model
      field :data, -> { Internal::Types::Array[Novabilling::Types::PlanOverrideResponse] }, optional: false, nullable: false
      field :meta, -> { Internal::Types::Hash[String, Object] }, optional: false, nullable: false
    end
  end
end
